test/integration/document_test.ts

```diff
@@ -761,6 +762,163 @@ describe('Document', function () {
+    await eventCollectorD3.waitAndVerifyNthEvent(1, {
+      status: DocumentStatus.Attached,
+      actorID: c1ID,
+    });
+
+    await c2.activate();
+    c2ID = c2.getID()!;
+    await c2.attach(d4, { syncMode: SyncMode.Manual });
+    await eventCollectorD4.waitAndVerifyNthEvent(1, {
+      status: DocumentStatus.Attached,
+      actorID: c2ID,
+    });
+
+    // 5. When c1 removes a document, it receives a removed event.
+    await c1.remove(d3);
+    await eventCollectorD3.waitAndVerifyNthEvent(2, {
+      status: DocumentStatus.Removed,
+    });
+
+    // 6. When c2 syncs, it should also receive a removed event.
+    await c2.sync();
+    await eventCollectorD4.waitAndVerifyNthEvent(2, {
+      status: DocumentStatus.Removed,
+    });
+
+    // 7. If the document is in the removed state, a detached event should not occur when deactivating.
+    const eventCount3 = eventCollectorD3.getLength();
+    const eventCount4 = eventCollectorD4.getLength();
+    await c1.deactivate();
+    await c2.deactivate();
+    assert.equal(eventCollectorD3.getLength(), eventCount3);
+    assert.equal(eventCollectorD4.getLength(), eventCount4);
+
+    unsub1();
+    unsub2();
+    unsub3();
+    unsub4();
+  });
+
+  it('Should properly handle removed document when deactivating', async function ({
+    task,
+  }) {
+    const c1 = new yorkie.Client(testRPCAddr);
+    const c2 = new yorkie.Client(testRPCAddr);
+    await c1.activate();
+    await c2.activate();
+    const c1ID = c1.getID()!;
+    const c2ID = c2.getID()!;
+
+    const docKey = toDocKey(`${task.name}-${new Date().getTime()}`);
+    const d1 = new yorkie.Document(docKey);
+    const d2 = new yorkie.Document(docKey);
+    const eventCollectorD1 = new EventCollector<StatusChangedEvent['value']>();
+    const eventCollectorD2 = new EventCollector<StatusChangedEvent['value']>();
+    const unsub1 = d1.subscribe('status', (event) => {
+      eventCollectorD1.add(event.value);
+    });
+    const unsub2 = d2.subscribe('status', (event) => {
+      eventCollectorD2.add(event.value);
+    });
+
+    // 1. When the client attaches a document, it receives an attached event.
+    await c1.attach(d1, {
+      syncMode: SyncMode.Manual,
+    });
+    await c2.attach(d2, {
+      syncMode: SyncMode.Manual,
+    });
+
+    await eventCollectorD1.waitAndVerifyNthEvent(1, {
+      status: DocumentStatus.Attached,
+      actorID: c1ID,
+    });
+    await eventCollectorD2.waitAndVerifyNthEvent(1, {
+      status: DocumentStatus.Attached,
+      actorID: c2ID,
+    });
+
+    // 2. When c1 removes a document, it receives a removed event.
+    await c1.remove(d1);
+    await eventCollectorD1.waitAndVerifyNthEvent(2, {
+      status: DocumentStatus.Removed,
+    });
+
+    // 3. When c2 deactivates, it should also receive a removed event.
+    await c2.deactivate();
+    // NOTE(chacha912): For now, document status changes to `Detached` when deactivating.
+    // This behavior may change in the future.
```

Comment:
@chacha912 
Would you explain your plan about this comment? 
Until `0.4.25-rc` of Yorkie Android sdk, it doesn't update status to detached when deactivating. 
I found it with yorkie-js-sdk's test code, and I'll add a commit for fix it. (But PR is not yet. I'll make it after apply 0.4.27~28.)
I made a consensus about this with @7hong13 

But, I found this comment now, I want to check this how it will be change future.

Reply:
@maruldy At that time, there was a possibility of reattaching a document after detaching it, and it hadn't been decided whether deactivate should perform detach. So I left a `NOTE`. 
If it's later decided that deactivate should include the same process of detaching the document, I considered that when c2 deactivates, it would receive the change of c1 removing the document, resulting in the final document status changing to `removed`. 
Currently, when deactivating, only `deactivateInternal` is performed internally, changing all document statuses to `detached`, so the final document status changes to `detached`.

Reply:
I think this thread is resolved via below conversation. Thanks.